### DISCUSSION TITLE
overlapping speech

### DISCUSSION THREAD

**Discussion by Majdoddin** [openai/whisper#434] [Q&A] [answered]
I have tried whisper on audios with overlapping speech, that is talking simultaneously. check it [**here** ](https://colab.research.google.com/github/Majdoddin/nlp/blob/main/whisper_source_seperation_change_of_speaker.ipynb). Whisper is very able to separate overlapping speech, but only generates transcription for one of them (I don't know on how it chooses one).  So my question is: _Is it an architectural limitation, that whisper has to ignore one of the overlapping speakers?_ Or can whisper be fine-tuned to generate transcriptions for both?

**Comment by Tetsujinfr** [comment]
Any chance for a comment OpenAI? @jongwook ?
thanks for the great repo/research.

**Reply by Tejasvidash** [reply]
I am working for speaker diarization where I need to build the model which will automatically detect the speakers. can you please help me with this?

**Comment by jongwook** [comment, marked as answer]
This is a limitation of the model because the training data often had transcription for one speaker while treating other voices as background noise. You might be able to nudge the model to produce overlapping transcription by prompting what each speaker would be saying 30 seconds ago, potentially with speaker labels like `--prefix "[Bob] So I was saying that [Alice] But there was always this"`. Please note, `[` and `]` are suppressed from the output by default, and you'll need to edit this line to re-enable them:

https://github.com/openai/whisper/blob/9f70a352f9f8630ab3aa0d06af5cb9532bd8c21d/whisper/tokenizer.py#L248

If you have a dataset of overlapping speech (or create one yourself by mixing single-speaker audio), I think fine-tuning on it can be a more effective/reliable/less hacky way to transcribe overlapping audio. This could be an interesting research project!

**Reply by loretoparisi** [reply]
@jongwook this is interresting! how you could prompt whisper about the specific speaker? Thank you
